src/main/java/ratpack/pac4j/RatpackPac4j.java

```diff
@@ -405,18 +397,15 @@ public static Handler authenticator(String path, ClientsProvider clientsProvider
    * @return a promise for the user profile
    * @see #userProfile(Context)
    */
-  public static <T extends CommonProfile> Promise<Optional<T>> userProfile(Context ctx, Class<T> type) {
+  public static <T extends UserProfile> Promise<Optional<T>> userProfile(Context ctx, Class<T> type) {
     return Promise.async(f ->
-      toProfile(type, f, ctx.maybeGet(CommonProfile.class), () ->
-        ctx.get(Session.class)
-          .get(Pac4jSessionKeys.USER_PROFILE)
-          .then(p -> {
-            if (p.isPresent()) {
-              ctx.getRequest().add(UserId.class, UserId.of(p.get().getId()));
-            }
-            toProfile(type, f, p, () -> f.success(Optional.<T>empty()));
-          })
-      )
+        toProfile(type, f, ctx.maybeGet(UserProfile.class), () ->
+            webContext(ctx).map((c) -> c.getProfileManager().getProfile())
```

Comment:
Why the switch to get the profile from the profile manager instead of getting it from the session?